internal/api/web/views/components/quiz_components/answers.templ

```diff
@@ -0,0 +1,56 @@
+package quiz_components
+
+import "github.com/Molnes/Nyhetsjeger/internal/data/questions"
+import "fmt"
+
+type Optional struct {
+	correctAnswered []string
+}
+
+type Option func(*Optional) Optional
+
+func CorrectAndAnswered(correct string, answered string) Option {
+	return func(o *Optional) Optional {
+		o.correctAnswered = append(o.correctAnswered, correct, answered)
+		return *o
+	}
+}
+
+var answeredCorrectly bool = false
+
+func isCorrectOrChosen(id string, option ...Option) string {
+	o := Optional{}
+	for _, opt := range option {
+		o = opt(&o)
+	}
+	for i := 0; i < len(o.correctAnswered); i += 2 {
+		if o.correctAnswered[i] == id {
+
+			if o.correctAnswered[i] == o.correctAnswered[i+1] && o.correctAnswered[i+1] == id {
+				answeredCorrectly = true
+			}
+
+			return "bg-green-500"
+		}
+		if o.correctAnswered[i+1] == id {
+			return "bg-red-500"
+		}
+	}
+	return "bg-white"
+}
```

Comment:
Again, document this more, or change how it works.
Or possibly split it up to be easier to understand and read.